src/uiprotect/api.py

```diff
@@ -825,8 +845,22 @@ async def update(self) -> Bootstrap:
             bootstrap = await self.get_bootstrap()
             self.__dict__.pop("bootstrap", None)
             self._bootstrap = bootstrap
+            await self.update_keyrings()
+            await self.update_ulpusers()
```

Comment:
```suggestion
            await self._update_keyrings()
            await self._update_ulpusers()
```

Reply:
We may need to add this into the bootstrap data so it can be updated from the websocket.

Reply:
Maybe we can overlay the data into `Bootstrap` with something like this

```diff
diff --git a/src/uiprotect/data/bootstrap.py b/src/uiprotect/data/bootstrap.py
index d8629c5..13512a8 100644
--- a/src/uiprotect/data/bootstrap.py
+++ b/src/uiprotect/data/bootstrap.py
@@ -181,6 +181,8 @@ class Bootstrap(ProtectBaseObject):
     sensors: dict[str, Sensor]
     doorlocks: dict[str, Doorlock]
     chimes: dict[str, Chime]
+    keyring: dict[str, Keyring]
+    ulp_users: dict[str, ULPUser]
     last_update_id: str
 
     # TODO:
```